### PR TITLE
Improve test environment

### PR DESCRIPTION
**Pull request by ihcomega56**
Added changes which improve execution environment.

Most updates are transferred from https://github.com/karolh2000/pyrsia-integration-tests/tree/karolh2000/docker_download_test, and they seem good. Thanks.

### CODE REVIEW

File: .github/workflows/run-bats-tests.yml [hunk 1]
```diff
@@ -7,16 +7,14 @@ on:
   pull_request:
     types:
       - opened
-      - closed
-      - synchronized
+      - synchronize
       - reopened
-      - ready_for_review
```
Comment (karolh2000) [inline .github/workflows/run-bats-tests.yml:13]: we might keep this one, as far as I understand this one is triggered when PR is switched from "draft" to "ready for review".

File: bats/tests/common-setup.bash [hunk 2]
```diff
@@ -13,27 +13,24 @@ _common_setup() {
 
 _common_setup_file() {
   echo "Setting up the test environment..." >&3
-  if [ -z "$GIT_REPO" ]; then
```
Comment (karolh2000) [inline bats/tests/common-setup.bash:16]: Not sure why we remove this code? As far as I understand it's responsible for picking up vars from the environment.

**Comment by karolh2000** [comment]
@ihcomega56  I think we shouldn't be merging [docker_download_test](https://github.com/karolh2000/pyrsia-integration-tests/tree/karolh2000/docker_download_test) as proposed in this PR. I think we should talk a little bit more about what is worth cherry-picking  from the mentioned branch and what it's not.  [docker_download_test](https://github.com/karolh2000/pyrsia-integration-tests/tree/karolh2000/docker_download_test) is experimental and contains things we don't want to push to the main branch. Basically, this branch wipes out some of the bug fixes and later changes so let's have a meeting and talk about it. Thanks!

File: readme.md [hunk 3]
```diff
@@ -33,26 +33,14 @@ In case you forgot to set `--recurse-submodules` during `clone` you can run the
 git submodule update --init
 ```
 
-Prerequisite:
```
Comment (karolh2000) [inline readme.md:36]: We don't want to remove this part of the docs.

File: readme.md [hunk 4]
```diff
@@ -33,26 +33,14 @@ In case you forgot to set `--recurse-submodules` during `clone` you can run the
...
-
 Run the tests:
 
+Prerequisite: Ensure that docker daemon is running and [JQ](https://stedolan.github.io/jq/) is installed.
+
 ```sh
 REPO_DIR=<path to your integration tests repo> $REPO_DIR/bats/run_tests.sh
 ```
 
-Optional variables for `run_tests.sh` script:
```
Comment (karolh2000) [inline readme.md:52]: Same here, we need this.

**Review by karolh2000 — CHANGES_REQUESTED** [review]

**Comment by ihcomega56** [comment]
@karolh2000 Thank you for the comment. Yes, I hope we can talk a bit more. I will tell why some codes can be removed as the other branch did.

File: bats/tests/common-setup.bash (hunk 2 above)
Comment (ihcomega56) [reply bats/tests/common-setup.bash:16]: I think we should fix Git commands (from line 23) or clean up $PYRSIA_TEMP_DIR every time to make $GIT_REPO and $GIT_BRANCH work. I shouldn't have removed it though...

File: readme.md (hunk 3 above)
Comment (ihcomega56) [reply readme.md:36]: Got it. I was misunderstanding a bit.

**Comment by ihcomega56** [comment]
I added some comments for the discussion next week. Thank you.

**Comment by karolh2000** [comment]
@ihcomega56 Closing PR and as we discussed please move the relevant parts to a new PR/branch. Thanks!

**Pull request closed without merging**